tests/core/test_migrate.py

```diff
@@ -764,3 +765,53 @@ def test_migrate_domain_raises_for_non_domain_files(tmp_path: Path):
         f"Please make sure to include these for a successful migration.",
     ):
         rasa.core.migrate.migrate_domain_format(domain_dir, domain_dir)
+
+
+def test_migrate_twice(tmp_path: Path,):
+    domain_dir = tmp_path / "domain"
+    domain_dir.mkdir()
+    migrated_domain_dir = tmp_path / "domain2"
+
+    prepare_domain_path(
+        domain_dir,
+        """
+        version: "2.0"
+        entities:
+            - outdoor
+        slots:
+          outdoor_seating:
+           type: bool
+           influence_conversation: false
+        """,
+        "slots_one.yml",
+    )
+
+    prepare_domain_path(
+        domain_dir,
+        """
+        version: "2.0"
+        entities:
+            - outdoor
+        slots:
+          cuisine:
+           type: text
+           influence_conversation: false
+        """,
+        "slots_two.yml",
+    )
+
+    with pytest.raises(
+        RasaException,
+        match="Domain files with multiple 'slots' sections were provided.",
+    ):
+        rasa.core.migrate.migrate_domain_format(domain_dir, migrated_domain_dir)
+
+    with pytest.raises(
+        RasaException,
+        match="Domain files with multiple 'slots' sections were provided.",
+    ):
+        rasa.core.migrate.migrate_domain_format(domain_dir, migrated_domain_dir)
+
+    current_dir = domain_dir.parent
+    assert not os.path.exists(current_dir / "original_domain")
+    assert not os.path.exists(current_dir / "new_domain")
```

Comment:
Since you provided `domain2` dir as out_path, I think `new_domain` never gets created. One thought here should we get also get rid of any migrated files in out_path dir if it's provided in case migration fails?

Reply:
Updated the test to check for multiple scenarios!

Reply:
Nice one! It's much clearer, but I have two follow-up q's:

- in the case of provided `migrated_domain_dir`, does the whole folder get removed instead of only its contents/files?
Removing the whole directory might be annoying for a user who might have created a specific location for this.

- in the case the default `domain_dir` is provided as `out_path`, what persists in this folder finally, only the original files I guess, meaning nothing changes there? 🤔

Reply:
1. Updated it and also updated the test to check if the dir was not removed (`assert Path.exists(migrated_domain_dir)` after the migration)
2. I guess so. I didn't touch this logic tbh, don't think it makes sense to do it as part of this issue.